### ISSUE TITLE
Managing multiple libraries

### ISSUE THREAD

**Issue by neacsum** [vector-of-bool/pitchfork#44]
PFL doesn't  address the integration of multiple libraries in one project. The document below offers a possible solution.
[The Dogfood and the Pitchfork.md](https://github.com/vector-of-bool/pitchfork/files/8607987/The.Dogfood.and.the.Pitchfork.md)

Thank you for taking the time to consider it.

**Comment by Quincunx271** [comment]
I think from your description that you want a way to set up dependencies within the pitchfork layout. There is such a way: `external/`. You can put the headers and source files there (or use a fit submodule, or use a symbolic link, etc.). If you really want precompiled lib.a files, they can go there too.

To use this, you can set up your build system to automatically set up include directories and link the libraries.

**Comment by neacsum** [comment]
Not exactly. If my library (`lib_A`) needs another library (`lib_B`) how will the consumer of my library (`lib_C`) know where to place the `lib_B`? Does he need to read my build instructions and figure out where everything goes? And now if all 3 libraries are used in an app, will I have to figure out what to do with each library?

My proposal wanted to offer a standard mechanism __between projects__, the same way PFL offers a standard for locations within a project.

**Comment by SiebrenW** [comment]
Isn't that what LD_LIBRARY_PATH and a FindMyLib.cmake is for?
If it's part of the toolchain and it's correctly installed for the compiler then you only need to add the name of the lib to the linker and that's it.
Say your application depends on the library in the README (or just rely on cmake crashing on find_package) and the user should know what's up.
No reason to make it more complicated.

**Comment by neacsum** [comment]
> Say your application depends on the library in the README (or just rely on cmake crashing on find_package) and the user should know what's up.
If life would only be that simple! You can run into different versions of libraries, libraries that depend on other libraries and so o,n and so on. For a more in-depth presentation of my suggestion, take a look at https://neacsu.net/docs/programming/code_layout/
